File: python/src/tools/insights/delete.py
import json
from pydantic import BaseModel
from src.tools.types import Context, Tool, ToolResult, TextContent


class DeleteInsightParams(BaseModel):
    insightId: int


async def delete_insight_handler(context: Context, params: DeleteInsightParams) -> ToolResult:
    project_id = await context.get_project_id()
    result = await context.api.insights(project_id).delete(params.insightId)

    if not result.success:
        raise Exception(f"Failed to delete insight: {result.error}")

    return ToolResult(content=[TextContent(text=json.dumps(result.data))])


def delete_insight_tool() -> Tool[DeleteInsightParams]:
    return Tool(
        name="insight-delete",
        description="Delete an insight by ID (soft delete - marks as deleted).",
        schema=DeleteInsightParams,
        handler=delete_insight_handler
    )

File: typescript/src/tools/surveys/delete.ts
import { SurveyDeleteSchema } from "@/schema/tool-inputs";
import { getToolDefinition } from "@/tools/toolDefinitions";
import type { Context, Tool } from "@/tools/types";
import type { z } from "zod";

const schema = SurveyDeleteSchema;
type Params = z.infer<typeof schema>;

export const deleteHandler = async (context: Context, params: Params) => {
	const { surveyId } = params;
	const projectId = await context.stateManager.getProjectId();

	const deleteResult = await context.api.surveys({ projectId }).delete({
		surveyId,
	});

	if (!deleteResult.success) {
		throw new Error(`Failed to delete survey: ${deleteResult.error.message}`);
	}

	return {
		content: [{ type: "text", text: JSON.stringify(deleteResult.data) }],
	};
};

const definition = getToolDefinition("survey-delete");

const tool = (): Tool<typeof schema> => ({
	name: "survey-delete",
	description: definition.description,
	schema,
	handler: deleteHandler,
	annotations: {
		destructiveHint: true,
		idempotentHint: false,
		openWorldHint: false,
		readOnlyHint: false,
	},
});

export default tool;
